lib/common/component.go

```diff
@@ -0,0 +1,76 @@
+package common
+
+import "github.com/blang/semver"
+
+// Component is the basic building block for all systems.
+// Schema info: https://github.com/opencontrol/schemas#component-yaml
+//
+// GetName returns the name of the component
+//
+// GetKey returns the key for the component (may not be unique). Useful for creating directories.
+//
+// SetKey sets the key for the component. Useful for overriding.
+//
+// GetAllSatisfies gets all the Satisfies objects for the component.
+//
+// GetVerifications get all the verifications.
+//
+// GetReferences get all the references.
+//
+// GetVersion returns the version
+//
+// SetVersion sets the version for the component.
+//
+// GetResponsibleRole gets the responsible party / role for the component.
```

Comment:
Is it typical to have this documentation up here rather than above each individual function?